backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/optimizations/LocalEscapeAnalysis.kt

```diff
@@ -74,6 +74,11 @@ internal object LocalEscapeAnalysis {
                         } ?: node.callee.irFunction?.isExternal?.let {
                             !(it || node.callee.irFunction.isBuiltInOperator)
                         } ?: true
+                        // Connect with all arguments that return value pointsTo.
+                        val returnPointsToMask = node.callee.pointsTo?.elementAtOrNull(node.callee.parameters.size) ?: 0
+                        if (returnPointsToMask and (1 shl index) != 0) {
```

Comment:
Here should be generic code connecting all possible nodes (see usage of `pointsTo` in global escape analysis).